aws_topology/stackstate_checks/aws_topology/resources/stepfunction.py

```diff
@@ -0,0 +1,344 @@
+get_activity_task
+get_execution_history
+"""
+
+
+StateMachineData = namedtuple('StateMachineData', ['state_machine', 'tags'])
+ActivityData = namedtuple('ActivityData', ['activity', 'tags'])
+
+
+class StepFunction(Model):
+    stateMachineArn = StringType(required=True)
+    definition = StringType(default="{}")
+    roleArn = StringType()
+
+
+class Activity(Model):
+    activityArn = StringType(required=True)
+    name = StringType()
+
+
+class StepFunctionCollector(RegisteredResourceCollector):
+    API = "stepfunctions"
+    API_TYPE = "regional"
+    COMPONENT_TYPE = "aws.stepfunction.statemachine"
+
+    @set_required_access_v2('states:ListTagsForResource', ignore=True)
+    def collect_tags(self, arn):
+        return self.client.list_tags_for_resource(resourceArn=arn).get('tags')
+
+    @set_required_access_v2('states:DescribeStateMachine', ignore=True)
+    def collect_state_machine_description(self, arn):
+        return self.client.describe_state_machine(stateMachineArn=arn)
+
+    def collect_state_machine(self, summary):
+        arn = summary.get('stateMachineArn')
+        state_machine = self.collect_state_machine_description(arn) or {'stateMachineArn': arn}
+        tags = self.collect_tags(arn) or []
+        return StateMachineData(state_machine=state_machine, tags=tags)
+
+    def collect_state_machines(self):
+        for state_machine in [
+            self.collect_state_machine(summary) for summary in
+            client_array_operation(
+                self.client,
+                'list_state_machines',
+                'stateMachines'
+            )
+        ]:
+            yield state_machine
+
+    def collect_activity(self, data):
+        activity_arn = data.get('activityArn')
+        tags = self.collect_tags(activity_arn)
+        return ActivityData(activity=data, tags=tags)
+
+    def collect_activities(self):
+        for activity in [
+            self.collect_activity(activity) for activity in
+            client_array_operation(
+                self.client,
+                'list_activities',
+                'activities'
+            )
+        ]:
+            yield activity
+
+    def process_all(self, filter=None):
+        if not filter or 'stepfunction' in filter:
+            self.process_state_machines()
+        if not filter or 'activities' in filter:
+            self.process_activities()
+
+    @set_required_access_v2('states:ListActivities', ignore=True)
+    def process_activities(self):
+        for activity in self.collect_activities():
+            self.process_activity(activity)
+
+    @set_required_access_v2('states:ListStateMachines', ignore=True)
+    def process_state_machines(self):
+        for state_machine in self.collect_state_machines():
+            self.process_state_machine(state_machine)
+
+    def process_activity(self, data):
+        output = make_valid_data(data.activity)
+        activity = Activity(data.activity, strict=False)
+        activity.validate()
```

Comment:
`.validate()` actually throws an exception, so you will probably have to start handling them or are they propagated and handled at the top?

Reply:
That is a thing I want to solve in a next branch on all APIs that use schematics validation. I want to solve it with a decorator.